src/staticsemantics/validateParameters.js

```diff
@@ -0,0 +1,110 @@
+// Copyright 2015 Traceur Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {ParseTreeVisitor} from '../syntax/ParseTreeVisitor.js';
+import {StringSet} from '../util/StringSet.js';
+import {isStrictKeyword} from '../syntax/Keywords.js';
+
+/**
+ * This visitor is used to find strict keywords (and eval/arguments) for
+ * strict function formal parameters as well as finding duplicate bindings in
+ * formal paramater list for strict functions and sloppy function with non
+ * simple formal parameters.
+ *
+ * In the case of non strict params we keep track of the duplicate names in case
+ * we see a default, rest or destructuring later in the list.
+ */
+class ParameterValidationVisitor extends ParseTreeVisitor {
+  /**
+   * @param {boolean} isStrict
+   * @param {ErrorReporter} reporter
+   */
+  constructor(isStrict, reporter) {
+    super();
+    this.reporter_ = reporter;
+    this.names_ = new StringSet();
+    this.errors_ = [];
+    this.reportStrictKeywords_ = isStrict;
+    this.reportDuplicates_ = isStrict;
```

Comment:
It's hard to sort out what will happen because `this.reportDuplicates_` is set both upon construction and dynamically. Did you consider a `ParameterValidationVisitor` for `isStrict === false` and a `StrictParameterValidationVisitor` for `true`?  Then `this.reportDuplicates_` only has one job.